tests/download/test_session.py

```diff
@@ -121,3 +121,35 @@ def test_session_sharing_multiprocess(fake_token, fake_config, memory_name):
         assert all(collected_session.token == fake_token for collected_session in collected_sessions)
     finally:
         thread.join()
+
+
+@pytest.mark.parametrize("memory_name", [None, "test-name"])
+def test_session_sharing_object(fake_token, fake_config, memory_name):
+    session = SentinelHubSession(config=fake_config, refresh_before_expiry=0, _token=fake_token)
+
+    kwargs = {} if memory_name is None else {"memory_name": memory_name}
+    manager = SessionSharing(session, name="thread name", **kwargs)
+
+    with pytest.raises(FileNotFoundError):
+        collect_shared_session(**kwargs)
+
+    with manager:
+        collected_session = collect_shared_session(**kwargs)
+        assert collected_session.token == fake_token
+
+    with pytest.raises(FileNotFoundError):
```

Comment:
do we want this function to raise a more informative error?
not sure, it feels like we're improving this way beyond what is needed/expected, but also it feels wrong to leave users confused :see_no_evil:

Reply:
Good point, I added that the same `FileNotFoundError` type gets raised but with a better error message.